app/src/main/java/org/simple/clinic/monthlyReports/questionnaire/BaseComponentDataPolymorphicJsonAdapter.kt

```diff
@@ -0,0 +1,25 @@
+package org.simple.clinic.monthlyReports.questionnaire
+
+import com.squareup.moshi.adapters.PolymorphicJsonAdapterFactory
+import org.simple.clinic.monthlyReports.questionnaire.component.BaseComponentData
+import org.simple.clinic.monthlyReports.questionnaire.component.ViewGroupComponentData
+import org.simple.clinic.monthlyReports.questionnaire.component.HeaderComponentData
+import org.simple.clinic.monthlyReports.questionnaire.component.InputFieldComponentData
+import org.simple.clinic.monthlyReports.questionnaire.component.LineSeparatorComponentData
+import org.simple.clinic.monthlyReports.questionnaire.component.SeparatorComponentData
+import org.simple.clinic.monthlyReports.questionnaire.component.SubHeaderComponentData
+import org.simple.clinic.monthlyReports.questionnaire.component.UnknownComponent
+
+class BaseComponentDataPolymorphicJsonAdapter {
+
+  fun getFactory(): PolymorphicJsonAdapterFactory<BaseComponentData> {
+    return PolymorphicJsonAdapterFactory.of(BaseComponentData::class.java, "view_type")
+        .withSubtype(ViewGroupComponentData::class.java, "view_group")
+        .withSubtype(HeaderComponentData::class.java, "header")
+        .withSubtype(SubHeaderComponentData::class.java, "sub_header")
+        .withSubtype(InputFieldComponentData::class.java, "input_field")
+        .withSubtype(SeparatorComponentData::class.java, "separator")
+        .withSubtype(LineSeparatorComponentData::class.java, "line_separator")
+        .withDefaultValue(UnknownComponent())
```

Comment:
Whenever there's an unknown component, can we raise an error to Sentry/Mixpanel so server-devs get alerted of malformed DSL?

Reply:
Unknown is in 3 places I think: BaseComponentDataPolymorphicJsonAdapter, Orientation & InputField

Reply:
@olttwa You are right. It's a big task. Will create a separate story for this.